Tests/UnitTests/Plugins/GeoModel/GeoDetectorObjectTest.cpp

```diff
@@ -22,41 +26,124 @@
+    }
+    std::vector<const Acts::Surface*> surfaces = box->surfaces();
+
+    for (auto surface : surfaces) {
+      const Acts::SurfaceBounds& sbounds = surface->bounds();
+      // check straws
+      if (surface->type() == Acts::Surface::SurfaceType::Straw) {
+        const auto* lineBounds =
+            dynamic_cast<const Acts::LineBounds*>(&sbounds);
+        BOOST_CHECK(geoDims.tube[1] == lineBounds->get(Acts::LineBounds::eR));
+        BOOST_CHECK(geoDims.tube[2] ==
+                    lineBounds->get(Acts::LineBounds::eHalfLengthZ));
+      }
+      // rectangle Surface check corner position without trf
+      if (sbounds.type() == Acts::SurfaceBounds::eRectangle) {
+        double csxmin = sbounds.values()[0];
+        double csymin = sbounds.values()[1];
+        double csxmax = sbounds.values()[2];
+        double csymax = sbounds.values()[3];
+        BOOST_CHECK(geoDims.boxI[0] == -csxmin);
+        BOOST_CHECK(geoDims.boxI[1] == -csymin);
+        BOOST_CHECK(geoDims.boxI[0] == csxmax);
+        BOOST_CHECK(geoDims.boxI[1] == csymax);
+      }
+      // trap Surface without trf
+      if (sbounds.type() == Acts::SurfaceBounds::eTrapezoid) {
+        const auto* trapBounds =
+            dynamic_cast<const Acts::TrapezoidBounds*>(&sbounds);
+        std::vector<Acts::Vector2> trapVerts = trapBounds->vertices();
+
+        for (long unsigned int i = 0; i < trapVerts.size(); i++) {
+          BOOST_CHECK(trapVerts[i][0] == geoDims.trapVerts[i][0]);
+          BOOST_CHECK(trapVerts[i][1] == geoDims.trapVerts[i][1]);
+        }
+      }
+    }
+  }
+}
+struct GeoGeometry {
+  std::vector<GeoFullPhysVol*> fpvs;
+  GeoDims dim;
+};
+GeoGeometry constructGeoModel() {
   // define materials
-  auto material = new GeoMaterial("Material", 1.0);
-  auto al = new GeoMaterial("Aluminium", 1.0);
+  GeoIntrusivePtr<GeoMaterial> material(new GeoMaterial("Material", 1.0));
+  GeoIntrusivePtr<GeoMaterial> al(new GeoMaterial("Aluminium", 1.0));
 
   // define dimensions
-  double gmhlx = 100, gmhly = 200, gmhlz = 2;
-  double gmrmin = 5, gmrmax = 5, gmhlzt = 100;
-  double gmhlxs = 100, gmhlys = 200, gmhlzs = 2;
+  GeoDims geoDims;
+  geoDims.boxO = {100, 200, 2};
+  geoDims.boxI = {100, 150, 1};
+  geoDims.trapVerts = {{-103, -50}, {103, -50}, {183, 50}, {-183, 50}};
+  geoDims.polyVerts = {{-60, -50}, {60, -50},  {153, 0},
+                       {123, 50},  {-123, 50}, {-153, 0}};
+  geoDims.tube = {5, 6, 100};
+  geoDims.poly_z = 2;
+  geoDims.trapHls = {
+      fabs(geoDims.trapVerts[0][0] - geoDims.trapVerts[1][0]) / 2,
+      fabs(geoDims.trapVerts[2][0] - geoDims.trapVerts[3][0]) / 2,
+      fabs(geoDims.trapVerts[0][1] - geoDims.trapVerts[2][1]) / 2};
 
   // create shapes
-  auto boxXY = new GeoBox(gmhlx, gmhly, gmhlz);
-  auto tube = new GeoTube(gmrmin, gmrmax, gmhlzt);
-  auto ssurface = new GeoBox(gmhlxs, gmhlys, gmhlzs);
+  auto boxXY = new GeoBox(geoDims.boxO[0], geoDims.boxO[1], geoDims.boxO[2]);
```

Comment:
I guess you can also use smart pointer for the shapes, if I am not wrong? with intrusive pointers again..